src/lib.rs

```diff
@@ -497,37 +497,37 @@ impl OffersMessageHandler for OfferHandler {
                     Ok(payment_id) => {
                         info!("Successfully verified invoice for payment_id {payment_id}");
                         let mut active_payments = self.active_payments.lock().unwrap();
-                        match active_payments.get_mut(&payment_id) {
-                            Some(pay_info) => match pay_info.invoice {
-                                Some(_) => {
-                                    error!("We already received an invoice with this payment id.")
-                                }
-                                None => {
-                                    pay_info.state = PaymentState::InvoiceReceived;
-                                    pay_info.invoice = Some(invoice.clone());
-                                }
-                            },
+                        let pay_info = match active_payments.get_mut(&payment_id) {
+                            Some(pay_info) => pay_info,
                             None => {
-                                error!("We received an invoice request for a payment id that we don't recognize or already paid: {payment_id:?}. We will ignore the invoice.");
+                                warn!("We received an invoice for a non active payment: {payment_id:?}. Invoice is ignored.");
+                                return None;
+                            }
+                        };
+                        match pay_info.invoice {
+                            Some(_) => {
+                                warn!("We already received an invoice with this payment id. Invoice is ignored.");
+                            }
+                            None => {
+                                pay_info.state = PaymentState::InvoiceReceived;
+                                pay_info.invoice = Some(invoice.clone());
                             }
                         }
-                        responder.map(|r| (OffersMessage::Invoice(invoice), r.respond()))
-                    }
-                    Err(()) => {
-                        error!("Invoice verification failed for invoice: {invoice:?}");
-                        responder.map(|r| {
-                            (
-                                OffersMessage::InvoiceError(InvoiceError::from_string(
-                                    String::from("invoice verification failure"),
-                                )),
-                                r.respond(),
-                            )
-                        })
+
+                        None
                     }
+                    Err(()) => responder.map(|r| {
```

Comment:
What's the change being made here? If it's unrelated to logging, let's put it in another commit.

Reply:
I removed the log, so it moved the responder to the top

Reply:
Oops, my bad, I think I was looking at the hmac line further down and it looks like it's in the correct commit :)